packaging/docker/Dockerfile

```diff
@@ -0,0 +1,72 @@
+# Dockerfile
+#
+# This source file is part of the FoundationDB open source project
+#
+# Copyright 2013-2018 Apple Inc. and the FoundationDB project authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+FROM ubuntu:18.04
+
+# Install dependencies
+
+RUN apt-get update && \
+	apt-get install -y wget=1.19.4-1ubuntu2 \
+	dnsutils=1:9.11.3+dfsg-1ubuntu1.2 && \
+	rm -r /var/lib/apt/lists/*  
+
+# Install FoundationDB Binaries
+
+ARG FDB_VERSION
+ARG FDB_WEBSITE=https://www.foundationdb.org
+
+RUN mkdir -p /var/fdb/tmp
+WORKDIR /var/fdb/tmp
+ADD $FDB_WEBSITE/downloads/$FDB_VERSION/linux/fdb_$FDB_VERSION.tar.gz .
+RUN tar -xzf fdb_$FDB_VERSION.tar.gz --strip-components=1
+RUN rm fdb_$FDB_VERSION.tar.gz
+RUN chmod u+x fdb*
+RUN mv fdb* /usr/bin
+WORKDIR /var/fdb
+RUN rm -r tmp
```

Comment:
Yes. That's the case because every RUN or any other command creates a new layer, which usually results in bigger overall image size. Just to illustrate:

```
...
ADD $FDB_WEBSITE/downloads/$FDB_VERSION/linux/fdb_$FDB_VERSION.tar.gz .
RUN tar -xzf fdb_$FDB_VERSION.tar.gz --strip-components=1
RUN rm fdb_$FDB_VERSION.tar.gz
RUN chmod u+x fdb*
RUN mv fdb* /usr/bin
WORKDIR /var/fdb
RUN rm -r tmp
...
```

Every line creates a new layer, meaning that even if have done `rm fdb_$FDB_VERSION.tar.gz` and `rm -r tmp`, the intermediate layers would still keep this files. This is effectively just a junk, because no one can access this file in the final image. Hence it's recommended to produce fewer number of layers by chaining commands:

```
RUN wget $FDB_WEBSITE/downloads/$FDB_VERSION/linux/fdb_$FDB_VERSION.tar.gz \
 && tar -xzf fdb_$FDB_VERSION.tar.gz --strip-components=1 \
 && rm fdb_$FDB_VERSION.tar.gz \
 && chmod u+x fdb* \
 && mv fdb* /usr/bin \
 && cd /var/fdb \
 && rm -r tmp
```

Reply:
I've restructured the dockerfile a bit to reduce the number of layers.

Reply:
I'm also thinking that in a future enhancement we may want to use a multistage build and just copy the binaries and libraries from an earlier stage into the final stage to reduce layering.